impl/src/main/java/com/okta/sdk/impl/resource/identity/provider/DefaultIdentityProviderBuilder.java

```diff
@@ -0,0 +1,285 @@
+                        .setSignature(client.instantiate(ProtocolAlgorithmTypeSignature.class)
+                            .setAlgorithm(responseSignatureAlgorithm)
+                            .setScope(responseSignatureScope))))
+                .setEndpoints(client.instantiate(ProtocolEndpoints.class)
+                    .setAcs(client.instantiate(ProtocolEndpoint.class)
+                        .setBinding(acsEndpointBinding)
+                        .setType(acsEndpointType))
+                    .setAuthorization(client.instantiate(ProtocolEndpoint.class)
+                        .setBinding(authorizationEndpointBinding)
+                        .setUrl(authorizationEndpointUrl))
+                    .setToken(client.instantiate(ProtocolEndpoint.class)
+                        .setBinding(tokenEndpointBinding)
+                        .setUrl(tokenEndpointUrl))
+                    .setUserInfo(client.instantiate(ProtocolEndpoint.class)
+                        .setBinding(userInfoEndpointBinding)
+                        .setUrl(userInfoEndpointUrl))
+                    .setJwks(client.instantiate(ProtocolEndpoint.class)
+                        .setBinding(jwksEndpointBinding)
+                        .setUrl(jwksEndpointUrl)))
+                .setScopes(scopes)
+                .setType(protocolType)
+            .setCredentials(client.instantiate(IdentityProviderCredentials.class)
+                .setClient(client.instantiate(IdentityProviderCredentialsClient.class)
+                    .setClientId(clientId)
+                    .setClientSecret(clientSecret)))
+            .setIssuer(client.instantiate(ProtocolEndpoint.class)
+                .setUrl(baseUrl)))
+            .setPolicy(client.instantiate(IdentityProviderPolicy.class)
+            .setAccountLink(client.instantiate(PolicyAccountLink.class)
+                .setAction(PolicyAccountLink.ActionEnum.AUTO)
+                .setFilter(null))
+            .setProvisioning(client.instantiate(Provisioning.class)
+                .setAction(Provisioning.ActionEnum.AUTO)
+                .setConditions(client.instantiate(ProvisioningConditions.class)
+                    .setDeprovisioned(client.instantiate(ProvisioningDeprovisionedCondition.class)
+                        .setAction(ProvisioningDeprovisionedCondition.ActionEnum.NONE))
+                    .setSuspended(client.instantiate(ProvisioningSuspendedCondition.class)
+                        .setAction(ProvisioningSuspendedCondition.ActionEnum.NONE)))
+                .setGroups(client.instantiate(ProvisioningGroups.class)
+                    .setAction(ProvisioningGroups.ActionEnum.NONE)))
+            .setMaxClockSkew(maxClockSkew)
+            .setSubject(client.instantiate(PolicySubject.class)
+                .setUserNameTemplate(client.instantiate(PolicyUserNameTemplate.class)
+                    .setTemplate(subjectTemplate))
+                .setMatchType(policySubjectMatchType))));
```

Comment:
Are all of these fields required?
For example, I would guess that the jwks config could be discovered from the `issuer`.  That means these fields could be `null`, and the payload would end up with an empty `jwks` block (which may or may not cause a problem).   This goes for much of the nested structure of this object.

I'm not saying this doesn't work, I haven't used this endpoint, but if we go this route, we should include test to validate that each valid permutation of the builder works as expected (with minimal settings). And a few that expect errors, (and make sure those errors are descriptive enough),

For example, If I want to create an IdP for Google, I likely only need the to set some sort of type ("google"), scopes, client id, and client secret. (The issue is static for most social providers, and Okta already knows them)

**NOTE:** I've only created IdPs from the UI, so I'm making some [possibly incorrect] assumptions.

Reply:
The builder `DefaultIdentityProviderBuilder` intends to cover only the generic OIDC case detailed at https://developer.okta.com/docs/reference/api/idps/#add-generic-openid-connect-identity-provider. For the sake of clarity, I would rename this class to `DefaultOIDCIdentityProviderBuilder` or something like that. 

Like you mentioned, there are several other IdP types listed at https://developer.okta.com/docs/reference/api/idps/#add-identity-provider. We could provide separate builder classes for each type in the interest of convenience and usability.

Reply:
Pushed rename changes mentioned above.